### PR TITLE
[MM-55332] Convert `./components/permissions_gates/channel_permission_gate.tsx` from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Summary

Convert `./components/permissions_gates/channel_permission_gate.tsx` from Class Component to Function Component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25361
Jira https://mattermost.atlassian.net/browse/MM-55332

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — CHANGES_REQUESTED** [review]
One minor improvement, but looks great otherwise!

File: webapp/channels/src/components/permissions_gates/channel_permission_gate/channel_permission_gate.tsx [hunk 1]
```diff
@@ -36,16 +36,14 @@ type Props = {
...
-
-        if (hasPermission && !invert) {
-            return children;
-        }
-        if (!hasPermission && invert) {
-            return children;
-        }
-        return null;
+const ChannelPermissionGate = ({hasPermission, children, invert = false}: Props) => {
+    if (hasPermission && !invert) {
```
[the remark this answers is not in the document]
Comment (Syed-Ali-Abbas-Zaidi) [reply webapp/channels/src/components/permissions_gates/channel_permission_gate/channel_permission_gate.tsx]: Done.

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @Syed-Ali-Abbas-Zaidi!

**Review by larkox — APPROVED** [review]
Great work! Thanks!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**